herbert_core/utilities/classes/@object_lookup/object_repmat.m

```diff
@@ -0,0 +1,177 @@
+%   iarray      Index or array of indices of object arrays to which to apply an
+%               implicit repmat operation. If not given, the operation is
+%               applied to all stored object arrays i.e. iarray = (1:obj.narray)
+%
+%   sz_repmat   One of:
+%                 - The size of a single array:   sz
+%                 - A cell array of array sizes: {sz1,sz2,...szN}
+%
+%               by which to implicitly replicate the subset of object arrays
+%               indicated by the input argument iarray using the Matlab function
+%               repmat. Here N is the number of indices in iarray.
+%
+%               The input object arrays are implicitly expanded as follows:
+%                 - single array size:
+%                       objArr1 => repmat(objArr1, sz)
+%                       objArr2 => repmat(objArr2, sz)
+%                           :               :
+%                 - cell array of array sizes:
+%                       objArr1 => repmat(objArr1, sz1)
+%                       objArr2 => repmat(objArr2, sz2)
+%                           :               :
+%               The arguments sz (or sz1, sz2,...) must be valid single argument
+%               input to the Matlab intrinsic function repmat i.e.               
+%               - Valid array size from the Matlab function called size i.e. row
+%                 vectors of at least two integers all of which must be greater
+%                 than or equal to zero.
+%               - A single positive integer n greater than or equal to zero.
+%                 This is equivalent to size vector [n,n].
+%
+%               NOTE:
+%               -----
+%               In the case when there is only one object_array in the
+%               object_lookup and sz_repmat is a cell array {sz1,sz2,...szN},
+%               the number of stored arrays is increased from 1 to N, with the
+%                   1st: repmat(objArr, sz1)
+%                   2nd: repmat(objArr, sz2)
+%                           :
+%                   Nth: repmat(objArr, szN)
+%
+%               This provides a mechanism to create an object_lookup with many
+%               identical stored arrays without the overhead of potentially
+%               expensive multiple checks of object equivalence
+%
+%               EXAMPLE (assuming obj contains just one object array)
+%                   object_repmat (obj, '-repeat', {1,1,1,1,1})
+%
+%
+% Output:
+% -------
+%   obj_out     Output obj_lookup object
+
+
+% Check validity of object_array
+% ------------------------------
+if ~isscalar(obj)
+    error('HERBERT:object_lookup:invalid_argument', ...
+        'Only operates on a single object_lookup (i.e. object must be scalar');
+end
+if ~obj.filled
+    error('HERBERT:object_lookup:invalid_argument', ...
+        'The object_lookup is not initialised')
+end
+
+% Parse input
+% -----------
+% Determine number and type OK
+narg = numel(varargin);
+if ~(narg==2 || narg==3) || ~ischar(varargin{end-1})
```

Comment:
This check for `char` is obscuring what you actually want to check. Check explicitly for `-repeat` here.

Reply:
Comment out of date given a change made to the syntax of object_lookup.